### PR TITLE
TCM-356 | Polish `Home` Banners and Promo

### PR DESCRIPTION
**Pull request by vespertineSiren**
### Summary

- Referred to [this figma](https://www.figma.com/file/LXa0oBt9aSNOk0YMAeSM8Q/TRX-Consumer?node-id=2764%3A80) as a point of reference for paddings/margins/etc
- Create new Banner/Promo responsemodels, backend calls, models, edit endpointmodel enums
- Add new events and complete function bodies in `Home`
- Refactored banner layout to a simpler view hierarchy. 

### Issue

[TCM-356](https://hyfnla.atlassian.net/browse/TCM-356)

### Video/Screenshot

| Old Banner | New Banner |
|----|----|
|old_banner|new_banner|

https://user-images.githubusercontent.com/27742105/119242267-6350a600-bb11-11eb-9f55-5f07beb9dfe1.mov

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/managers/BackendManager.kt [hunk 1]
```diff
@@ -82,6 +87,11 @@ class BackendManager(private val api: BaseApi, private val cacheManager: CacheMa
         return call(RequestModel(endpoint = EndpointModel.PAYMENT_DELETE, path, params = null))
     }
 
+    suspend fun promos(): ResponseModel {
+        val path = EndpointModel.PROMOTIONS.path
+        return call(RequestModel(endpoint = EndpointModel.PROMOTIONS, path = path, params = null))
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/managers/BackendManager.kt:92]: we might want to rename the endpoint model to `promos`

**Comment by onodude** [comment]
thank you @vespertineSiren looking good

File: app/src/main/java/com/trx/consumer/managers/BackendManager.kt (hunk 1 above)
Comment (vespertineSiren) [reply app/src/main/java/com/trx/consumer/managers/BackendManager.kt]: Here you are: https://github.com/hyfn/trx-consumer-android/pull/57/commits/db0d02a0677829934184896e81df305498028683

**Review by VivekV95 — APPROVED** [review]

**Review by onodude — APPROVED** [review]
thank you!

**Pull request merged**